community/kernel/src/test/java/org/neo4j/kernel/api/impl/labelscan/LabelScanStoreTest.java

```diff
@@ -0,0 +1,640 @@
+import org.neo4j.helpers.collection.BoundedIterable;
+import org.neo4j.helpers.collection.Iterators;
+import org.neo4j.helpers.collection.PrefetchingIterator;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.kernel.api.labelscan.LabelScanStore;
+import org.neo4j.kernel.api.labelscan.LabelScanWriter;
+import org.neo4j.kernel.api.labelscan.NodeLabelRange;
+import org.neo4j.kernel.api.labelscan.NodeLabelUpdate;
+import org.neo4j.kernel.impl.api.scan.LabelScanStoreProvider.FullStoreChangeStream;
+import org.neo4j.kernel.lifecycle.LifeSupport;
+import org.neo4j.storageengine.api.schema.LabelScanReader;
+import org.neo4j.test.rule.RandomRule;
+import org.neo4j.test.rule.TestDirectory;
+import org.neo4j.test.rule.fs.DefaultFileSystemRule;
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.Matchers.startsWith;
+import static org.hamcrest.core.IsCollectionContaining.hasItem;
+import static org.hamcrest.core.IsCollectionContaining.hasItems;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.neo4j.collection.primitive.PrimitiveLongCollections.EMPTY_LONG_ARRAY;
+import static org.neo4j.helpers.collection.Iterators.iterator;
+import static org.neo4j.helpers.collection.Iterators.single;
+import static org.neo4j.io.fs.FileUtils.deleteRecursively;
+import static org.neo4j.kernel.api.labelscan.NodeLabelUpdate.labelChanges;
+
+public abstract class LabelScanStoreTest
+{
+    private final TestDirectory testDirectory = TestDirectory.testDirectory();
+    private final ExpectedException expectedException = ExpectedException.none();
+    private final DefaultFileSystemRule fileSystemRule = new DefaultFileSystemRule();
+    final RandomRule random = new RandomRule();
+
+    @Rule
+    public final RuleChain ruleChain = RuleChain.outerRule( random ).around( testDirectory ).around( expectedException )
+            .around( fileSystemRule );
+
+    private static final long[] NO_LABELS = new long[0];
+
+    private LifeSupport life;
+    private TrackingMonitor monitor;
+    private LabelScanStore store;
+    private File dir;
+
+    @Before
+    public void clearDir() throws IOException
+    {
+        dir = testDirectory.directory( "lucene" );
+        if ( dir.exists() )
+        {
+            deleteRecursively( dir );
+        }
+    }
+
+    @After
+    public void shutdown()
+    {
+        life.shutdown();
+    }
+
+    protected abstract LabelScanStore createLabelScanStore( FileSystemAbstraction fileSystemAbstraction,
+            File rootFolder, List<NodeLabelUpdate> existingData, boolean usePersistentStore, boolean readOnly,
+            LabelScanStore.Monitor monitor );
+
+    @Test
+    public void failToRetrieveWriterOnReadOnlyScanStore() throws IOException
```

Comment:
`failToRetrieveWriterOnReadOnlyScanStore` and `failToGetWriterOnReadOnlyScanStore` are the same